src/main/java/de/tum/in/www1/artemis/service/exam/ExamAccessService.java

```diff
@@ -80,6 +86,84 @@ public StudentExam getExamInCourseElseThrow(Long courseId, Long examId) {
         return studentExam.get();
     }
 
+    /**
+     * Retrieves a specified studentExam for a TestExam from the database and sends it to the client
+     * @param courseId the course to which the exam belongs
+     * @param examId the examId of the exam we are interested in
+     * @param studentExamId the id of the studentExam we are interested in
+     * @return a StudentExam without Exercises
+     */
+    public StudentExam getStudentExamForTestExamElseThrow(Long courseId, Long examId, Long studentExamId) {
+        User currentUser = userRepository.getUserWithGroupsAndAuthorities();
+
+        // Check that the current user is at least student in the course.
+        Course course = courseRepository.findByIdElseThrow(courseId);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, null);
```

Comment:
Since you already fetched the user, you could pass it into the method. Otherwise it has to be retrieved again